sparkmagic/sparkmagic/livyclientlib/reliablehttpclient.py

```diff
@@ -62,18 +63,20 @@ def _send_request_helper(self, url, accepted_status_codes, function, data, retry
                 error = True
                 r = None
                 status = None
+                text = None
 
                 self.logger.error(u"Request to '{}' failed with '{}'".format(url, e))
             else:
                 error = False
                 status = r.status_code
+                text = r.text
 
             if error or status not in accepted_status_codes:
                 if self._retry_policy.should_retry(status, error, retry_count):
                     sleep(self._retry_policy.seconds_to_sleep(retry_count))
                     retry_count += 1
                     continue
                 else:
-                    raise HttpClientException(u"Invalid status code '{}' or error '{}' from {}"
-                                              .format(status, error, url))
+                    raise HttpClientException(u"Invalid status code '{}' or error '{}' from {} with text {}"
```

Comment:
>with text

Somehow, I don't think that's clear enough. Maybe make it "with payload"?

Can you show an example of how r.text looks like for these type of exceptions?